tests/e2e/configurer/chain/commands.go

```diff
@@ -105,3 +113,23 @@ func (n *NodeConfig) CreateWallet(walletName string) string {
 	n.LogActionF("created wallet %s, waller address - %s", walletName, walletAddr)
 	return walletAddr
 }
+
+func (n *NodeConfig) QueryPropStatusTimed(proposalNumber int, desiredStatus string, totalTime chan time.Duration) {
+	start := time.Now()
+	require.Eventually(
+		n.t,
+		func() bool {
+			status, err := n.QueryPropStatus(proposalNumber)
+			if err != nil {
+				return false
+			}
+
+			return status == desiredStatus
+		},
+		1*time.Minute,
+		10*time.Millisecond,
+		"Osmosis node failed to retrieve prop tally",
+	)
+	elapsed := time.Since(start)
+	totalTime <- elapsed
```

Comment:
This is related to an earlier comment:
https://github.com/osmosis-labs/osmosis/pull/2341/files#r944726827

Currently, the problem, is that we can fail after trying for 1 minute. However, nothin is sent on the `totalTime` channel if the failure does occur.

As a result, the test can get blocked for a long period of time (I think 30 minutes max)

I think we have 2 options to mitigate that and fail immediately:
1. Create a second timeout specifically for that test case (e.g. 2 minute timeout). So that if we block on the channel, the test case fails early

2. have a `defer` statement that either sends `elapsed` on the channel or a dummy value (i.e. -1)signifying failure.

Please let me know what you think

Reply:
I went back and forth between your two suggestion and decided to implement it through a case timeout (similar to the recommendation from Go by Example [here](https://gobyexample.com/timeouts))

I added a sleep to the timed go routine to ensure it stopped at the desired time and it did! I think this was the simplest implementation possible. Lmk what you think